packages/plugin-nested-docs/src/utilities/shouldPopulateBreadcrumbs.ts

```diff
@@ -0,0 +1,47 @@
+import type { CollectionConfig } from 'payload/types'
+
+import type { PluginConfig } from '../types.js'
+
+/**
+ * Helper function to determine if the breadcrumbs should be populated
+ *
+ * @param pluginConfig - The plugin configuration
+ * @param data - The data to be saved
+ * @param collection - The collection configuration
+ * @param originalDoc - The original document
+ */
+export const shouldPopulateBreadcrumbs = (
+  pluginConfig: PluginConfig,
+  data: any,
+  collection: CollectionConfig,
+  originalDoc?: any,
+) => {
+  // the originalDoc is not present, so we should populate the breadcrumbs
+  if (!originalDoc) return true
+
+  // if it's a nested collection, we should populate the breadcrumbs
+  if (data?.hierarchy && !Array.isArray(data?.hierarchy)) {
```

Comment:
What is `data.hierarchy` here?

Reply:
sorry, it should take the value from the `breadcrumbsFieldSlug` from the config